src/Config/Converters/EntityGraphQLOptionsConverter.cs

```diff
@@ -0,0 +1,127 @@
+
+namespace Azure.DataApiBuilder.Config.Converters;
+
+internal class EntityGraphQLOptionsConverter : JsonConverter<EntityGraphQLOptions>
+{
+    /// <inheritdoc/>
+    public override EntityGraphQLOptions? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType == JsonTokenType.StartObject)
+        {
+            string singular = string.Empty;
+            string plural = string.Empty;
+            bool enabled = true;
+            GraphQLOperation? operation = null;
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    return new EntityGraphQLOptions(singular, plural, enabled, operation);
+                }
+
+                if (reader.TokenType == JsonTokenType.PropertyName)
+                {
+                    string? property = reader.GetString();
+                    reader.Read();
+
+                    switch (property)
+                    {
+                        case "enabled":
+                            enabled = reader.GetBoolean();
+                            break;
+                        case "type":
+                            if (reader.TokenType == JsonTokenType.String)
+                            {
+                                singular = reader.DeserializeString() ?? string.Empty;
+                            }
+                            else if (reader.TokenType == JsonTokenType.StartObject)
+                            {
+                                while (reader.Read())
+                                {
+                                    if (reader.TokenType == JsonTokenType.EndObject)
+                                    {
+                                        break;
+                                    }
+
+                                    if (reader.TokenType == JsonTokenType.PropertyName)
+                                    {
+                                        string? property2 = reader.GetString();
+                                        reader.Read();
+                                        switch (property2)
+                                        {
+                                            case "singular":
+                                                singular = reader.DeserializeString() ?? string.Empty;
+                                                break;
+                                            case "plural":
+                                                plural = reader.DeserializeString() ?? string.Empty;
+                                                break;
+                                        }
+                                    }
+                                }
+                            }
+
+                            break;
+
+                        case "operation":
+                            string? op = reader.DeserializeString();
```

Comment:
how do we deal with failures here or no case matched?

Reply:
Ignored. These are the only properties in the JSON we know about, if you add others they'll be ignored until the CLI does a new serialization, at which point in time they are dropped.

Reply:
which should help enable backwards (and forwards) compatibility.